src/chat/runeMagicCard.ts

```diff
@@ -0,0 +1,411 @@
+  }
+
+  public static async inputChangeHandler(ev: Event, messageId: string): Promise<void> {
+    const chatMessage = getGame().messages?.get(messageId);
+    const flags = chatMessage?.data.flags.rqg as RuneMagicCardFlags;
+
+    const form = (ev.target as HTMLElement).closest("form") as HTMLFormElement;
+    const formData = new FormData(form);
+
+    //@ts-ignore formData.entries
+    for (const [name, value] of formData.entries()) {
+      if (name in flags.formData) {
+        //@ts-ignore Type 'any' is not assignable to type 'never'.
+        flags.formData[name as keyof typeof flags.formData] = value;
+      }
+    }
+
+    //@ts-ignore Type 'AbstractEmbeddedItem' is missing the following properties from type 'RuneDataSource'
+    const selectedRune: RuneDataSource = flags.eligibleRunes.find(
+      //@ts-ignore _id
+      (i) => i._id === flags.formData.selectedRuneId
+    ) as Rune;
+
+    const newChance: number =
+      Number(selectedRune.data.chance) +
+      Number(flags.formData.ritualOrMeditation) +
+      Number(flags.formData.skillAugmentation) +
+      Number(flags.formData.otherModifiers);
+
+    flags.formData.chance = newChance;
+    const data = await RuneMagicCard.renderContent(flags);
+    await chatMessage?.update(data);
+  }
+
+  public static async formSubmitHandler(
+    ev: JQueryEventObject,
+    messageId: string
+  ): Promise<boolean> {
+    ev.preventDefault();
+
+    const chatMessage = getGame().messages?.get(messageId);
+    const flags = chatMessage?.data.flags.rqg as RuneMagicCardFlags;
+
+    const formData = new FormData(ev.target as HTMLFormElement);
+    // @ts-ignore formData.entries()
+    for (const [name, value] of formData.entries()) {
+      if (name in flags.formData) {
+        //@ts-ignore TODO: WHY?!
+        flags.formData[name as keyof typeof flags.formData] = value;
+      }
+    }
+
+    // @ts-ignore submitter
+    const button = ev.originalEvent.submitter as HTMLButtonElement;
+    button.disabled = true;
+    setTimeout(() => (button.disabled = false), 1000); // Prevent double clicks
+
+    const actor = getActorFromIds(flags.actorId, flags.tokenId);
+    if (actor) {
+      const speakerName = getSpeakerName(flags.actorId, flags.tokenId);
+      await RuneMagicCard.roll(flags.itemData, flags, actor, speakerName);
+    } else {
+      ui.notifications?.warn(getGame().i18n.localize("RQG.RuneMagicCard.getActorFromIdsWarning"));
+    }
+    return false;
+  }
+
+  public static async directRoll(
+    runeMagicItem: RqgItem,
+    actor: RqgActor,
+    speakerName: string
+  ): Promise<void> {
+    assertItemType(runeMagicItem?.data.type, ItemTypeEnum.RuneMagic);
+    const cult = actor.items.get(runeMagicItem.data.data.cultId);
+    assertItemType(cult?.data.type, ItemTypeEnum.Cult);
+    
+    let usableRuneNames: string[] = [];
+    let runesForCasting: RuneDataSource[] = [];
+    if (runeMagicItem.data.data.runes.includes(getGame().settings.get("rqg", "magicRuneName") as string)) {
```

Comment:
The settings can be typed [here](https://github.com/sun-dragon-cult/fvtt-system-rqg/blob/1400dec6fdde2d9b39b2b1eb86cfd2ac59dc9155/src/global.d.ts#L30) so that no `as string` is needed.

Reply:
I figured there must be a way to type that!